### ISSUE TITLE
ical - undefined method `gsub' for nil:NilClass

### ISSUE THREAD

**Issue by muhammad-usman-arshad** [icalendar/icalendar#215]
I am facing following error when I call 'to_ical' method. Can somebody help on this issue?

```
NoMethodError: undefined method `gsub' for nil:NilClass
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/value.rb:57:in `escape_param_value'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/value.rb:51:in `param_ical'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/value.rb:29:in `block in params_ical'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/value.rb:29:in `each'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/value.rb:29:in `map'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/value.rb:29:in `params_ical'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/value.rb:24:in `to_ical'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:43:in `block in ical_properties'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:35:in `map'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:35:in `ical_properties'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:26:in `to_ical'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:63:in `block (2 levels) in ical_components'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:62:in `each'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:62:in `block in ical_components'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:60:in `each'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:60:in `ical_components'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/bundler/gems/icalendar-b699701bc998/lib/icalendar/component.rb:27:in `to_ical'
	from (irb):113
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/railties-4.0.5/lib/rails/commands/console.rb:90:in `start'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/railties-4.0.5/lib/rails/commands/console.rb:9:in `start'
	from /home/dev/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/railties-4.0.5/lib/rails/commands.rb:62:in `<top (required)>'
	from bin/rails:4:in `require'
	from bin/rails:4:in `<main>'irb(main):114:0> cal.class
```

I am using following code to generate Icalendar Object.

```
cal = Icalendar::Calendar.new
cal.ip_method = 'REQUEST'
cal.event do |e|
	e.uid 		= event['iCalUID']
	e.created       = Icalendar::Values::DateTime.new(event['created'].utc, 'tzid' => tzid)
	e.last_modified = Icalendar::Values::DateTime.new(event['updated'].utc, 'tzid' => tzid)
	e.dtstart       = Icalendar::Values::DateTime.new(event['start']['dateTime'].utc, 'tzid' => tzid)
	e.dtend         = Icalendar::Values::DateTime.new(event['end']['dateTime'].utc, 'tzid' => tzid)
	e.summary       = event['summary']
	description_event	= (event['description']  rescue '') || "" 
		
	e.description   = description_event
	e.location      = event['location'] rescue ''
	e.organizer 		= Icalendar::Values::CalAddress.new("mailto:#{event['organizer']['email']}", :cn => event['organizer']['displayName'])
	e.transp				= 'OPAQUE'
	e.ip_class      = 'PRIVATE'
	e.sequence			= 0
	e.status				= 'CONFIRMED'

	e.attendee      = []
	event['attendees'].each do |attendee|
		e.attendee << Icalendar::Values::CalAddress.new("mailto:#{attendee.email}",
			:cutype => 'INDIVIDUAL',
			:role => 'REQ-PARTICIPANT',
			:partstat => attendee.responseStatus,
			:cn => !attendee.display_name.nil? ? attendee.display_name : ''
		)
	end
end
cal.to_ical
```

Following is the Object generated using above code. (Enter are added in following output for better visibility)

```
#<Icalendar::Event:0x00557f8e2159e0 @name="event", @ical_name="VEVENT", @custom_components={}, @custom_properties={}, 
@dtstamp=Wed, 07 Aug 2019 05:51:23 UTC +00:00, @uid="cqcgql95keq62miprb0m8unu58@google.com", 
@parent=#<Icalendar::Calendar:0x00557f8e4b1e88 @name="calendar", @ical_name="VCALENDAR", @custom_components={}, @custom_properties={}, 
@prodid="icalendar-ruby", @version="2.0", @calscale="GREGORIAN", @ip_method="REQUEST", @events=[#<Icalendar::Event:0x00557f8e2159e0 ...>]>, 
@created=Wed, 07 Aug 2019 05:24:31 UTC +00:00, @last_modified=Wed, 07 Aug 2019 05:24:31 UTC +00:00, 
@dtstart=Wed, 07 Aug 2019 14:45:00 UTC +00:00, @dtend=Wed, 07 Aug 2019 15:00:00 UTC +00:00, @summary="Test Staging", 
@description="To reschedule<br><a href = http://dev.meeting:5000/reschedulings?permalink=batman&slug=1301565155471>https://dev.meeting:5000/reschedulings/batman/1301565155471</a><br>To cancel<br><a href=\"http://dev.meeting:5000/cancellations?permalink=batman&slug=1301565155471\">https://dev.meeting:5000/cancellations/batman/1301565155471</a><br><a href='<%= reschedulings_url(slug: meeting.slug) %>' class='cancel-reschedule-url' target='_blank'>https://dev.meeting:5000/reschedulings/<%= meeting.slug %></a><br><br><a href='<%= cancellations_url(slug: meeting.slug) %>' class='cancel-reschedule-url' target='_blank'>https://dev.meeting:5000/cancellations/<%= meeting.slug %></a>", 
@location="Lahore", @organizer=#<URI::MailTo:0x00557f8e22dce8 URL:mailto:batman@gmail.com>, @transp="OPAQUE", @ip_class="PRIVATE", 
@sequence=0, @status="CONFIRMED", @attendee=[#<URI::MailTo:0x00557f8e236c30 URL:mailto:batman@gmail.com>, 
#<URI::MailTo:0x00557f8e235cb8 URL:mailto:superman@gmail.com>]>
```

**Comment by rahearn** [comment]
Thanks for your report.

Can you share the value of the `event` object, or at least where that is generated?

**Comment by muhammad-usman-arshad** [comment]
Trying to generate event on Google Calendar and calendar event object is already pasted in issue.

**Comment by rahearn** [comment]
What version of the gem are you using?

**Comment by muhammad-usman-arshad** [comment]
Version 2.2.2

**Comment by rahearn** [comment]
Thanks. That bug was fixed in version 2.4.0. Please use that or any more recent version.
